pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -108,27 +108,70 @@ func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	logger = logger.WithValues("deployment", dep.Name)
 
-	ing := &netv1.Ingress{}
-	ing.Name = util.FindOwnerKind(spc.OwnerReferences, "Ingress")
-	ing.Namespace = req.Namespace
-	logger = logger.WithValues("ingress", ing.Name)
-	if ing.Name != "" {
-		logger.Info("getting owner ingress")
-		if err = p.client.Get(ctx, client.ObjectKeyFromObject(ing), ing); err != nil {
+	return p.upsertObjectDeployment(dep, spc, req, ctx, logger)
+}
+
+func (p *PlaceholderPodController) placeholderPodCleanCheck(obj client.Object) (bool, error) {
+	nic, ok := obj.(*v1alpha1.NginxIngressController)
+	if ok {
+		if nic.Spec.DefaultSSLCertificate.KeyVaultURI == nil {
+			return true, nil
+		}
+		return false, nil
+	}
+	ing, ok := obj.(*netv1.Ingress)
+	if ok {
+		managed, err := p.ingressManager.IsManaging(ing)
+		if err != nil {
+			return false, fmt.Errorf("determining if ingress is managed: %w", err)
+		}
+		if ing.Name == "" || ing.Spec.IngressClassName == nil || !managed {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	return false, nil
+}
+
+func (p *PlaceholderPodController) upsertObjectDeployment(dep *appsv1.Deployment, spc *secv1.SecretProviderClass, req ctrl.Request, ctx context.Context, logger logr.Logger) (ctrl.Result, error) {
+	var (
+		err error
+		obj client.Object
+	)
+
+	result := ctrl.Result{}
+
+	if objName := util.FindOwnerKind(spc.OwnerReferences, "NginxIngressController"); objName != "" {
+		obj = &v1alpha1.NginxIngressController{}
+		obj.SetName(objName)
+	}
+
+	if objName := util.FindOwnerKind(spc.OwnerReferences, "Ingress"); objName != "" {
+		obj = &netv1.Ingress{}
+		obj.SetName(objName)
+		obj.SetNamespace(req.Namespace)
+	}
```

Comment:
these should be a switch statement with a default of returning an error. we need to handle the case of not being able to find the owner kind